include/zephyr/sw_isr_table.h

```diff
@@ -87,6 +94,25 @@ extern struct z_shared_isr_table_entry z_shared_sw_isr_table[];
  */
 unsigned int z_get_sw_isr_table_idx(unsigned int irq);
 
+/**
+ * @brief Helper function used to get the parent interrupt controller device based on passed IRQ.
+ *
+ * @param irq IRQ number in its zephyr format
+ *
+ * @return corresponding interrupt controller device in _sw_isr_table
+ */
+const struct device *z_get_sw_isr_device_from_irq(unsigned int irq);
```

Comment:
did not do any extensive review, but this caught my attention, we should avoid declaring such function in public headers and give the impression they are public APIs anyone can use. Please move those to where they are being used or putting them in some private header.